examples/inpainting/train_inpainting.py

```diff
@@ -0,0 +1,1362 @@
+        base_model=args.pretrained_model_name_or_path,
+        model_description=model_description,
+        inference=True,
+    )
+
+    tags = ["stable-diffusion", "stable-diffusion-diffusers", "inpainting", "diffusers"]
+    model_card = populate_model_card(model_card, tags=tags)
+
+    model_card.save(os.path.join(repo_folder, "README.md"))
+
+
+def log_validation(validation_dataloader, vae, text_encoder, tokenizer, unet, args, accelerator, weight_dtype, epoch):
+    logger.info("Running validation... ")
+
+    pipeline = StableDiffusionInpaintPipeline.from_pretrained(
+        args.pretrained_model_name_or_path,
+        vae=accelerator.unwrap_model(vae),
+        text_encoder=accelerator.unwrap_model(text_encoder),
+        tokenizer=tokenizer,
+        unet=accelerator.unwrap_model(unet),
+        safety_checker=None,
+        revision=args.revision,
+        variant=args.variant,
+        torch_dtype=weight_dtype,
+    )
+
+    pipeline = pipeline.to(accelerator.device)
+    pipeline.set_progress_bar_config(disable=True)
+
+    if args.enable_xformers_memory_efficient_attention:
+        pipeline.enable_xformers_memory_efficient_attention()
+
+    if args.seed is None:
+        generator = None
+    else:
+        generator = torch.Generator(device=accelerator.device).manual_seed(args.seed)
+
+    images = []
+    prompts = []
+    masks = []
+    image_transform = transforms.ToPILImage()
+    for _, batch in enumerate(validation_dataloader):
+        mask = image_transform(batch["masks"][0] * 255)
+        init_image = image_transform(batch["pixel_values"][0])
+        prompt = batch["prompts"][0]
+
+        with torch.autocast("cuda"):
+            #### UPDATE PIPELINE HERE
+            image = pipeline(
+                prompt, image=init_image, mask_image=mask, num_inference_steps=20, generator=generator
+            ).images[0]
+
+        prompts.append(prompt)
+        images.append(image)
+        masks.append(mask)
+
+    for tracker in accelerator.trackers:
+        if tracker.name == "tensorboard":
+            np_images = np.stack([np.asarray(img) for img in images])
+            tracker.writer.add_images("validation", np_images, epoch, dataformats="NHWC")
+        elif tracker.name == "wandb":
+            tracker.log(
+                {
+                    "validation": [
+                        wandb.Image(make_image_grid([image, masks[i]], 1, 2), caption=f"{i}: {prompts[i]}")
+                        for i, image in enumerate(images)
+                    ]
+                }
+            )
+        else:
+            logger.warn(f"image logging not implemented for {tracker.name}")
+
+    del pipeline
+    torch.cuda.empty_cache()
+
+    return images
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Simple example of a training script.")
+    parser.add_argument(
+        "--input_perturbation", type=float, default=0, help="The scale of input perturbation. Recommended 0.1."
+    )
```

Comment:
(nit): Seems like a data-related argument which we can place with the other data related arguments, no? Also, an expanded help would be nice. I don't know what the perturbation correspond to.

Reply:
I took it from the `text_to_image` training as it is. Sure I can move it.